homework/java/BuhtiyarovP_java_hw_2_1/src/test/java/by/jrr/service/LoginServiceTest.java

```diff
@@ -0,0 +1,142 @@
+package by.jrr.service;
+
+import by.jrr.been.User;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+public class LoginServiceTest {
+
+    LoginService loginService;
+    User user;
+    User blockedUser;
+    String positiveUserInput = "password";
+    String negativeUserInput = "wrong";
+
+    public LoginServiceTest() {
+    }
+
+    @Before
+    public void setUp() {
+        this.loginService = new LoginService();
+        this.user = getUser();
+        this.blockedUser = getBlockedUser();
+
+        Assert.assertEquals(3, user.getLoginAttempts());
+        Assert.assertFalse(user.isBlocked());
+
+        Assert.assertEquals(0, blockedUser.getLoginAttempts());
+        Assert.assertTrue(blockedUser.isBlocked());
+    }
+
+    @Test
+    public void checkUserPassword_positive() {
+
+        boolean actionResult = this.loginService.login(user, positiveUserInput);
+        Assert.assertTrue(actionResult);
+    }
+
+    @Test
+    public void checkUserPassword_negative() {
+
+        boolean actionResult = this.loginService.login(user, negativeUserInput);
+        Assert.assertFalse(actionResult);
+    }
+
+    @Test
+    public void reduceLoginAttempts(){
+        loginService.login(user,negativeUserInput);
+        Assert.assertEquals(2, user.getLoginAttempts());
+    }
+
+    @Test
+    public void login_positive(){
+
+        boolean actionResult = this.loginService.login(user, positiveUserInput);
+        Assert.assertTrue(actionResult);
+    }
+
+    @Test
+    public void login_negative(){
+
+        boolean actionResult = this.loginService.login(user, negativeUserInput);
+        Assert.assertFalse(actionResult);
+        Assert.assertEquals(2, user.getLoginAttempts());
+    }
+
+    @Test
+    public void blockUser(){
+        user.setLoginAttempts(1);
+        loginService.login(user,negativeUserInput);
+        Assert.assertTrue(user.isBlocked());
+    }
+
+    @Test
+    public void after3WrongPassword_UserIsBlocked(){
```

Comment:
I think not to use numbers in names methods and values better, because numbers isn't often visible